diff_drive_controller/src/diff_drive_controller.cpp

```diff
@@ -472,8 +472,20 @@ controller_interface::CallbackReturn DiffDriveController::on_configure(
       odometry_publisher_);
 
   auto & odometry_message = realtime_odometry_publisher_->msg_;
-  odometry_message.header.frame_id = odom_params_.odom_frame_id;
-  odometry_message.child_frame_id = odom_params_.base_frame_id;
+
+  std::string controller_namespace = std::string(node_->get_namespace());
```

Comment:
```suggestion
  std::string controller_namespace = std::string(get_node()->get_namespace());
```
I think it's something like this in the new api